pyxem/signals/indexation_results.py

```diff
@@ -147,6 +159,487 @@ def _get_second_best_phase(z):
+        annotation_shift: Sequence[float] = None,
+        text_kwargs: dict = None,
+        include_intensity: bool = False,
+        intesity_scale: float = 1,
+        **kwargs,
+    ):
+        """Convert the orientation map to a set of markers for plotting.
+
+        Parameters
+        ----------
+        n_best: int
+            The amount of solutions to plot
+        annotate : bool
+            If True, the euler rotation and the correlation will be annotated on the plot using
+            the `Texts` class from hyperspy.
+        marker_color: str, optional
+            The color of the point markers used for simulated reflections
+        text_color: str, optional
+            The color used for the text annotations for reflections. Does nothing if `annotate` is `False`.
+        annotation_shift: List[float,float], optional
+            The shift to apply to the annotation text. Default is [0,-0.1]
+        include_intensity: bool
+            If True, the intensity of the diffraction spot will be displayed with more intense peaks
+            having a larger marker size.
+        """
+        if text_kwargs is None:
+            text_kwargs = dict()
+        if annotation_shift is None:
+            annotation_shift = [0, -0.15]
+        if not self._lazy:
+            navigation_chunks = (5,) * self.axes_manager.navigation_dimension
+        else:
+            navigation_chunks = None
+
+        for n in range(n_best):
+            vectors = self.to_single_phase_vectors(
+                lazy_output=lazy_output, navigation_chunks=navigation_chunks
+            )
+            color = marker_colors[n % len(marker_colors)]
+            if include_intensity:
+                intensity = vectors.map(
+                    vectors_to_intensity,
+                    scale=intesity_scale,
+                    inplace=False,
+                    ragged=True,
+                    output_dtype=object,
+                    output_signal_size=(),
+                    navigation_chunks=navigation_chunks,
+                ).data.T
+                kwargs["sizes"] = intensity
+
+            coords = vectors.map(
+                vectors_to_coordinates,
+                inplace=False,
+                ragged=True,
+                output_dtype=object,
+                output_signal_size=(),
+                navigation_chunks=navigation_chunks,
+            )
+            markers = hs.plot.markers.Points.from_signal(
+                coords, facecolor="none", edgecolor=color, **kwargs
+            )
+            yield markers
+
+            if annotate:
+                texts = vectors.map(
+                    vectors_to_text,
+                    inplace=False,
+                    lazy_output=lazy_output,
+                    ragged=True,
+                    output_dtype=object,
+                    output_signal_size=(),
+                )
+                coords.map(lambda x: x + annotation_shift, inplace=True)
```

Comment:
Is there a good argument for using yield here? 

This probably has to be refactored eventually once I get around to making a `hs.compute()`.  Ideally we would go:

Create a big list of signals --> mass compute with hs.compute which merges the task graphs --> Make a list of markers from the computation.

For the lazy markers, we need to adjust how hyperspy handles plotting of markers.

Currently each marker is computed seperately and the signal is computed seperately as well where we should just merge the task graphs into one.  That would require a context manager --> `to_compute` attribute for hyperexplorer and then that gets checked/ run on each action.

@ericpre  I think we talked about this when first creating the lazy markers. 

https://github.com/hyperspy/hyperspy/blob/b742845d7f606bc4086f5bec4bc0ca84b8e4104d/hyperspy/drawing/markers.py#L504C3-L526

Reply:
I used yield to hopefully reduce memory usage, but using lazy markers would mitigate this